app/src/main/java/com/example/currencyconverter/CurrencyConverterApplication.kt

```diff
@@ -1,15 +1,12 @@
 package com.example.currencyconverter
 
 import android.app.Application
-import com.example.currencyconverter.domain.repository.CurrencyRepository
-import com.example.currencyconverter.ui.di.ServiceLocator
+import com.example.currencyconverter.domain.repository.RepositoryImpl
+import com.example.currencyconverter.di.ServiceLocator
 
 class CurrencyConverterApplication : Application() {
 
-    val currencyRepository: CurrencyRepository
+    val currencyRepository: RepositoryImpl
```

Comment:
Only the service locator should know about particular class implementation. Others should use its interface only.
Why do you actually need a repository link in the App class?